### DISCUSSION TITLE
How to create one search field to search/filter 3 columns of data

### DISCUSSION THREAD

**Discussion by lohhb** [Budibase/budibase#14382] [Help] [answered]
How can I create one search field that can search or filter 3 columns of data.
The search need to show one whole row of data if any field one of the 3 columns meant the requirement..

**Comment by mikesealey** [comment, marked as answer]
Hey @lohhb I've been looking into this for you

The Table Component allows the user to use the column heading as a search box by hovering over the title of the column and clicking the magnifying glass. Not relevant to this specific case, but still a cool feature you might not have known about.
![image](https://github.com/user-attachments/assets/ed3d51ce-3329-4842-9696-e3ce93b9c23a)

Now to the question you asked - here's how I approached this using the sample data provided with the Budibase InternalDB

On your screen, create a Form component, and add a text field. Apply the settings you see fit such as field name, placeholder text, etc. I have added mine into a container for some additional styling features.
![image](https://github.com/user-attachments/assets/ded9952a-12e1-436c-bd7c-3cafdb2e18a7)

![image](https://github.com/user-attachments/assets/d0453f2f-eb27-42f1-9088-7fe0749480e9)

In the settings of your text field, add an "On Change" action to Update State with the field value. Each time a change is made inside the field (ie the user types) the search query is updated. 
![image](https://github.com/user-attachments/assets/6dbfc012-6ede-4e13-b11c-50ab26d562f1)

Next, add a table component underneath, displaying the data you want to search. Add filters on the columns you want to be able to search, set to be "Like" the binding of the state that you set above. Importantly the behaviour is set to match _any_ filter.
![image](https://github.com/user-attachments/assets/805e1604-5da0-4589-95e1-d2cfd1c40d64)

Now when you type in the field, the search query is passed to the table which will filter all rows that have that text in any of the specified columns. The "Like" operator means that it doesn't have to be an exact match, simple that it contains those characters - "LL" will return Richard **Kelley** and **Holly Carroll** AND Francis Castro from **Wells*.
![image](https://github.com/user-attachments/assets/1a283bce-e5fb-4910-a18a-2fc5d2f2c7ab)

Hopefully this was easy enough to follow along, but if you wanted to take a closer look, I've attached the app-export of my work.
[14382-export-1723729963609.tar.gz](https://github.com/user-attachments/files/16625872/14382-export-1723729963609.tar.gz) You can read more about how to import an app [here](https://docs.budibase.com/docs/export-and-import-apps).

If there's anything that's still unclear I'll do my best to answer any questions you have.

**Comment by lohhb** [comment]
Thank you very much time and reply, it works!
